### ISSUE TITLE
Parsing existing Document, two different styles seems identical

### ISSUE THREAD

**Issue by ZvikaZ** [python-openxml/python-docx#248]
Hi.
Please see the attached file.
[snippet1.docx](https://github.com/python-openxml/python-docx/files/100483/snippet1.docx)

It has two lines (in Hebrew), first line is size 11 in Word, second line is size 10 in Word.
When parsing it with the following code:

    import docx
    
    doc_file_name = 'snippet1.docx'
    word_doc = docx.Document(doc_file_name)
    
    runs = []
    for (paragraph) in word_doc.paragraphs:
        for (run) in paragraph.runs:
            if run.text.strip():
                print run.text, run.style.style_id, "bold:", run.bold, "size:", run.font.size
    
It prints:

     אב  s01 bold: None size: None
     האהבה העליונה  s01 bold: None size: None

I tried to compare both 'run's in debugger, and couldn't find anything that tells the size difference between them.

Please help...
Thanks,
Zvika

**Comment by scanny** [comment]
What's your question?

**Comment by ZvikaZ** [comment]
Well, I assume that the size difference is hiding somewhere, and I couldn't
find it, so where should it be?

Or, maybe it's some corner case that Docx doesn't show that data, so I'd
like to report it.

Thanks!
Zvika
בתאריך 22 בינו׳ 2016 20:17,‏ "Steve Canny" <notifications@github.com> כתב:

**Comment by ZvikaZ** [comment]
Hi.
I think I can narrow down my question...

I've splitted the file to 2 files, each containing one line:
[snippet1a.docx](https://github.com/python-openxml/python-docx/files/102200/snippet1a.docx)
[snippet1b.docx](https://github.com/python-openxml/python-docx/files/102201/snippet1b.docx)

Unzipped the .docx's, and tried to diff them.

My issue seems to be here:

File 1:

```xml
	<w:r>
		<w:rPr>
			<w:rStyle w:val="s01"/>
			<w:rFonts w:hint="cs"/>
			<w:rtl/>
		</w:rPr>
		<w:t xml:space="preserve">אב </w:t>
	</w:r>
```

File 2:

```xml

	<w:r>
		<w:rPr>
			<w:rStyle w:val="s01"/>
			<w:rFonts w:hint="cs"/>
			<w:szCs w:val="20"/>
			<w:rtl/>
		</w:rPr>
		<w:t xml:space="preserve">האהבה העליונה </w:t>
	</w:r>
```

It seems that the difference is in `szCs`. 
I don't have enough experience with .docx format, does it make sense that `szCs` would cause my case?

Grepping in the `docx` code, found `szCs` at `./oxml/text/font.py`, but it seems to be not used.
Am I on the correct lead?

Thanks again,
Zvika

**Comment by scanny** [comment]
The .size property reports only a *directly applied* size. If the displayed size come from higher up the style hierarchy you'll need to traverse that hierarchy to get it.

"cs" in this context stands for *cursive script*, of which Hebrew might be one. Arabic certainly is. They have a separate size attribute from the Latin typefaces, I expect so they can be made to look natural in a paragraph where one is embedded among the other.

**Comment by ZvikaZ** [comment]
Indeed, this attribute is what I need.
In the meanwhile, I tried to implement support for it.
It's quite clunky, it'd be great if you have the time to do something better with your knowledge of python-docx...

Attaching my 'git diff' output as some starting point:
[diff.txt](https://github.com/python-openxml/python-docx/files/104786/diff.txt)

Thanks for the help,
Zvika

**Comment by scanny** [comment]
That actually looks pretty good @ZvikaZ, definitely enough to go on once we get to the implementation. Thanks for posting it back once you got it working :)

**Comment by ZvikaZ** [comment]
Hi.
Great to hear that it looks pretty good.
Was it incorporated into your code?

**Comment by scanny** [comment]
No. Getting a merge requires both acceptance and unit tests and documentation updates.

But snippets like yours are a good resource for a developer who is developing a feature. Also they can be handy for folks who need to add the capability but don't have the skills to contribute committable code.

**Comment by ZvikaZ** [comment]
Do you have an estimation if and (roughly) when it will be merged?

**Comment by ZvikaZ** [comment]
Hi again,
It'd be really great if this can be addressed...

Thanks

**Comment by ZvikaZ** [comment]
Hi.
I've came back to this :-)

Are there any news with it?
If not, is PR welcomed?
